packages/react-scripts/config/webpack.config.js

```diff
@@ -383,22 +355,50 @@ module.exports = function (webpackEnv) {
             // https://github.com/jshttp/mime-db
             {
               test: [/\.avif$/],
-              loader: require.resolve('url-loader'),
-              options: {
-                limit: imageInlineSizeLimit,
-                mimetype: 'image/avif',
-                name: 'static/media/[name].[hash:8].[ext]',
+              type: 'asset',
+              mimetype: 'image/avif',
+              parser: {
+                dataUrlCondition: {
+                  maxSize: imageInlineSizeLimit,
+                },
               },
             },
             // "url" loader works like "file" loader except that it embeds assets
             // smaller than specified limit in bytes as data URLs to avoid requests.
             // A missing `test` is equivalent to a match.
             {
               test: [/\.bmp$/, /\.gif$/, /\.jpe?g$/, /\.png$/],
-              loader: require.resolve('url-loader'),
-              options: {
-                limit: imageInlineSizeLimit,
-                name: 'static/media/[name].[hash:8].[ext]',
+              type: 'asset',
+              parser: {
+                dataUrlCondition: {
+                  maxSize: imageInlineSizeLimit,
+                },
+              },
+            },
+            {
+              test: /\.svg$/,
+              use: [
+                {
+                  loader: '@svgr/webpack',
+                  options: {
+                    prettier: false,
+                    svgo: false,
+                    svgoConfig: {
+                      plugins: [{ removeViewBox: false }],
+                    },
+                    titleProp: true,
+                    ref: true,
+                  },
+                },
+                {
+                  loader: 'file-loader',
+                  options: {
+                    name: 'static/media/[name].[hash].[ext]',
+                  },
+                },
+              ],
```

Comment:
@dawnofmidnight  I'm using the file-loader in combination with the svgr/loader - do you have a suggestion to change this to the asset module pattern? (maybe we could add a task to update it later to get the alpha out?)

Reply:
I haven't messed with SVGR much, so I'll take a look and see if I can find something out. Adding a task for later is probably the best option unless someone else knows something about using asset modules instead.

Reply:
ok, I remember working on this and using the asset module pattern, but the tests broke so I kept the file-loader - might need a bit more effort not sure if @evenstensberg can help out here or have some insights? :)

Reply:
Without `file-loader` or `url-loader`, svgr usage changes:
```diff
-import starUrl, { ReactComponent as Star } from './star.svg';
+import Star from './star.svg';
```

One solution I found is using [`new URL()`](https://webpack.js.org/guides/asset-modules/#url-assets) calls. These calls are processed by asset modules and can return output similar to `file-loader` or `url-loader` depending on the asset type. You can specify `dependency: { not: ['url'] }` to exclude `new URL()` calls from being processed by loaders and let asset modules handle it.

Based on this, I made a small loader [`new-url-loader`](https://github.com/marella/new-url-loader#readme) that generates `new URL()` calls and can provide the functionality of both `file-loader` and `url-loader` using asset modules. For example, you can replace `file-loader` using:

```diff
{
  test: /\.svg$/,
- use: ['@svgr/webpack', 'file-loader'],
+ oneOf: [
+   {
+     dependency: { not: ['url'] },
+     use: ['@svgr/webpack', 'new-url-loader'],
+   },
+   {
+     type: 'asset/resource',
+   },
+ ],
}
```

Now you can continue to use both the URL and React component:

```js
import starUrl, { ReactComponent as Star } from './star.svg';
```

I added some [tests](https://github.com/marella/new-url-loader/blob/main/test/index.test.js) that check if its output matches with `file-loader` and `url-loader` and also created a [minimal example](https://github.com/marella/new-url-loader/tree/main/examples/svgr).

Hope it helps.

Reply:
Thanks @marella, I'm not fully convinced I would like to see a "type" instead of the url/file loader otherwise I don't see the big difference (I could be missing out on something)

Let's defer this challenge into a follow up issue+pr. (to unblock the initial release)

I'll mark as resolved for now and opened an issue to address and talki about this further #11213 @dawnofmidnight, @marella and others please feel free to subscribe :)